From b871e15a92adb20b3d66fdf80272b29a13fdf971 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:08:22 -0400
Subject: [usb] sierra: remove bogus endpoint test

Message-id: <1410887463-20674-431-git-send-email-dzickus@redhat.com>
Patchwork-id: 95548
O-Subject: [RHEL7 PATCH 430/591] USB: sierra: remove bogus endpoint test
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 21aa1c41dbe820ec7d2271f3e675eb1be2320245
Author: Johan Hovold <jhovold@gmail.com>
Date:   Mon May 26 19:22:59 2014 +0200

    USB: sierra: remove bogus endpoint test

    Remove bogus endpoint-address test which is never true.

    Signed-off-by: Johan Hovold <jhovold@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/serial/sierra.c b/drivers/usb/serial/sierra.c
index 4cb11b7..169899f 100644
--- a/drivers/usb/serial/sierra.c
+++ b/drivers/usb/serial/sierra.c
@@ -747,9 +747,6 @@ static struct urb *sierra_setup_urb(struct usb_serial *serial, int endpoint,
  struct urb *urb;
  u8  *buf;
 
- if (endpoint == -1)
-  return NULL;
-
  urb = usb_alloc_urb(0, mem_flags);
  if (!urb)
   return NULL;
-- 
1.7.1